card/mod/01_core/set/all/phases.rb

```diff
@@ -206,3 +246,33 @@ def when_condition_applies? block
 def success
   Env.success(cardname)
 end
+
+def prepare_for_phases
+  identify_action
+  reset_patterns
+  include_set_modules
+end
+
+def run_phase?
+  director.main?
+end
+
+def validation_phase
+  # TODO: try to use Card.current_act as condition in callback definition
+  return true unless run_phase?
+  director.validation_phase
+end
+
+def storage_phase &block
+  director.storage_phase &block
```

Comment:
Ambiguous block operator. Parenthesize the method arguments if it's surely a block operator, or add a whitespace to the right of the `&` if it should be a binary AND.